library/core/src/array/mod.rs

```diff
@@ -364,3 +364,54 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as self, with `f` applied to each element.
+    ///
+    /// # Examples
+    /// ```
+    /// #![feature(array_map)]
+    /// let x = [1,2,3];
+    /// let y = x.map(|v| v + 1);
+    /// assert_eq!(y, [2,3,4]);
+    /// ```
+    #[unstable(feature = "array_map", issue = "77777")]
+    pub fn map<F, S>(self, mut f: F) -> [S; N]
+    where
+        F: FnMut(T) -> S,
```

Comment:
Should this be `U` rather than `S`? Most of the other `map` uses `U` except `Iterator::map` which uses `B`.

Reply:
Ah no, I just picked S randomly, will change